src/Taxonomy_Command.php

```diff
@@ -148,12 +148,22 @@ protected function get_counts( $taxonomies ) {
 	public function list_( $args, $assoc_args ) {
 		$formatter = $this->get_formatter( $assoc_args );
 
+		// Check if it's strict mode or not.
+		$is_strict = ( isset( $assoc_args['strict'] ) && true === $assoc_args['strict'] ) ? true : false;
+
+		unset( $assoc_args['strict'] );
+
 		if ( isset( $assoc_args['object_type'] ) ) {
 			$assoc_args['object_type'] = array( $assoc_args['object_type'] );
+			$taxonomy_object           = $assoc_args['object_type'];
+		} else {
+			$taxonomy_object = get_post_types();
 		}
 
 		$fields     = $formatter->fields;
-		$taxonomies = get_taxonomies( $assoc_args, 'objects' );
+		$taxonomies = ( $is_strict ) ?
+			get_taxonomies( $assoc_args, 'objects' ) :
+			get_object_taxonomies( $taxonomy_object, 'objects' );
```

Comment:
`$taxonomy_object` might not be defined here. So, further up we need to run a check to see if we have an object type if `--strict` was defined, and throw a `WP_CLI::error()` if not.

Reply:
```suggestion
			: get_object_taxonomies( $taxonomy_object, 'objects' );
```

Reply:
> `$taxonomy_object` might not be defined here. So, further up we need to run a check to see if we have an object type if `--strict` was defined, and throw a `WP_CLI::error()` if not.

No, I need to correct myself. Throwing an error is not wanted here. What we want is to fall back to the previous default behavior if no object is provided. `get_object_taxonomies()` does not support this, so the entire condition should be reversed and extended:

```php
$taxonomies = ( isset( $taxonomy_object ) && ! $strict )
			? get_object_taxonomies( $taxonomy_object, 'objects' );
			: get_taxonomies( $assoc_args, 'objects' );
```